src/test/unit_test/multi_runner.h

```diff
@@ -0,0 +1,354 @@
+
+public:
+    multi_runner_base();
+    ~multi_runner_base();
+
+    std::size_t
+    checkout_test_index();
+
+    std::size_t
+    checkout_job_index();
+
+    void
+    any_failed(bool v);
+
+    void
+    add(results const& r);
+
+    void
+    inc_keep_alive_count();
+
+    std::size_t
+    get_keep_alive_count();
+
+    template <class S>
+    void
+    print_results(S& s);
+
+    bool
+    any_failed() const;
+};
+
+}  // detail
+
+//------------------------------------------------------------------------------
+
+/** Manager for children running unit tests
+ */
+class multi_runner_parent : private detail::multi_runner_base</*IsParent*/true>
+{
+private:
+    // message_queue_ is used to collect log messages from the children
+    std::ostream& os_;
+    std::atomic<bool> continue_message_queue_{true};
+    std::thread message_queue_thread_;
+
+public:
+    multi_runner_parent(multi_runner_parent const&) = delete;
+    multi_runner_parent&
+    operator=(multi_runner_parent const&) = delete;
+
+    multi_runner_parent();
+    ~multi_runner_parent();
+
+    bool
+    any_failed() const;
+};
+
+//------------------------------------------------------------------------------
+
+/** A class to run a subset of unit tests
+ */
+class multi_runner_child : public beast::unit_test::runner,
+                           private detail::multi_runner_base</*IsParent*/ false>
+{
+private:
+    std::size_t job_index_;
+    detail::results results_;
+    detail::suite_results suite_results_;
+    detail::case_results case_results_;
+    std::size_t num_jobs_{0};
+    bool quiet_{false};
+    bool print_log_{true};
+
+    std::atomic<bool> continue_keep_alive_{true};
+    std::thread keep_alive_thread_;
+
+public:
+    multi_runner_child(multi_runner_child const&) = delete;
+    multi_runner_child&
+    operator=(multi_runner_child const&) = delete;
+
+    multi_runner_child(std::size_t num_jobs, bool quiet, bool print_log);
+    ~multi_runner_child();
+
+    template <class Pred>
+    bool
+    run_multi(Pred pred);
+
+private:
+    virtual void
+    on_suite_begin(beast::unit_test::suite_info const& info) override;
+
+    virtual void
+    on_suite_end() override;
+
+    virtual void
+    on_case_begin(std::string const& name) override;
+
+    virtual void
+    on_case_end() override;
+
+    virtual void
+    on_pass() override;
+
+    virtual void
+    on_fail(std::string const& reason) override;
+
+    virtual void
+    on_log(std::string const& s) override;
+};
+
+//------------------------------------------------------------------------------
+
+template <class Pred>
+bool
+multi_runner_child::run_multi(Pred pred)
+{
+    auto const& suite = beast::unit_test::global_suites();
+    auto const num_tests = suite.size();
+    // actual order to run the tests. Use this to move longer running tests to
+    // the beginning to better take advantage of a multi process run.
+    std::vector<std::size_t> order(num_tests);
```

Comment:
I toyed with making this more generic to allow you to just paste in a list of tests.  In case you like it: https://github.com/bachase/rippled/commit/884d2d3ca52c98d6870339b24d9160150f7a5d55.

Reply:
@bachase I like what you did here. I simplified it a little, take a look at the "Resolve Feedback Issues 1" commit to see if it's OK now.